### DISCUSSION TITLE
source/DialogueBox.hx:109: characters 58-66 : Expected )

### DISCUSSION THREAD

**Discussion by AshuraTheEchidna** [FunkinCrew/Funkin#1562] [Compiling help]
here is the line 109, comment if you can help me:
if (PlayState.SONG.song.tolowerCase() -- 'senpai' || PlayState.SONG.song.tolowerCase() -- 'roses' || PlayState.SONG.song.tolowerCase() -- 'thorns')

**Comment by ghost** [comment]
get latest haxe from website

**Comment by AshuraTheEchidna** [comment]
> get latest haxe from website

i already have it, and yes, i tried re-installing

**Comment by NotHyper-474** [comment]
Weren't the `--` supposed to be `==`? I've searched if Haxe has a `--` operator but couldn't find nothing about it.

**Reply by NotHyper-474** [reply]
Turns out Haxe did, in fact, have a `--` operator. This still should've been `==` instead though.

**Comment by ghost** [comment]
Statement.''ninjamuffin99".company

**Comment by ghost** [comment]
> Weren't the `--` supposed to be `==`? I've searched if Haxe has a `--` operator but couldn't find anything about it.

its supposed to be  `==`  because `--` would make it break the game right?

**Comment by ghost** [comment]
Hi - Try `==` instead of `--` and post the outcome here.

**Comment by ghost** [comment]
> Weren't the `--` supposed to be `==`? I've searched if Haxe has a `--` operator but couldn't find nothing about it.

[there is a -- operator you dumb dumb](https://haxe.org/manual/expression-operators-unops.html)

**Comment by ghost** [comment]
thats minus though
2 non-bool values will not make a bool when subtracted

**Comment by NotHyper-474** [comment]
> > Weren't the `--` supposed to be `==`? I've searched if Haxe has a `--` operator but couldn't find nothing about it.
> 
> [there is a -- operator you dumb dumb](https://haxe.org/manual/expression-operators-unops.html)

That one only works used as `insertNumberHere--` tho.
